File: psets/1/mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // Get user input
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    // Start printing
    for (int i = 1; i <= height; i++)
    {
        int spaces = height - i;
        for (int s = 0; s < spaces; s++)
        {
            printf(" ");
        }

        for (int hashes = 0; hashes < i; hashes++)
        {
            printf("#");
        }

        printf("\n");
    }
}

File: psets/6/sentimental-mario-more/mario.py
#!/usr/bin/env python3

def get_int(prompt):
    while True:
        try:
            n = int(input(prompt))
            if (n <= 0 or n > 8):
                continue
            return n
        except ValueError:
            pass


def print_pyramid(n):
    for i in range(n):
        spaces = " " * (n - (i + 1))
        hashes = "#" * (i + 1)
        print(f"{spaces}{hashes}  {hashes}")


def main():
    height = get_int("Height: ")
    print_pyramid(height)


main()
